projects/client/RabbitMQ.Client/src/client/impl/ModelBase.cs

```diff
@@ -1231,8 +1250,52 @@ public void BasicCancel(string consumerTag)
                 mandatory,
                 basicProperties,
                 body);
-        }
+        }
+
+        public void BasicBatchPublish(string exchange,
+    string routingKey,
+    bool mandatory,
+    IEnumerable<BatchMessage> messages)
+        {
+            foreach (var message in messages)
+            {
+                if (message.basicProperties == null)
+                {
+                    message.basicProperties = CreateBasicProperties();
+                }
+
+                if (NextPublishSeqNo > 0)
+                {
+                    lock (m_unconfirmedSet.SyncRoot)
+                    {
+                        if (!m_unconfirmedSet.Contains(NextPublishSeqNo))
+                        {
+                            m_unconfirmedSet.Add(NextPublishSeqNo);
+                        }
+                        NextPublishSeqNo++;
+                    }
+                }
+            }
 
+            _Private_BasicBatchPublish(exchange,
```

Comment:
These methods are produced by the ApiGen project, not added manually.

Reply:
This is where we all need a little help.
I don't know if this is considered Protocol specific or Client API specific.